kad/src/protocol.rs

```diff
@@ -262,13 +272,22 @@ fn proto_to_msg(mut message: protobuf_structs::dht::Message) -> Result<KadMsg, I
                 Ok(KadMsg::FindNodeReq {
                     key: message.take_key(),
                 })
+
             } else {
+                let mut closer_peers = Vec::with_capacity(message.get_closerPeers().len());
+                for peer in message.mut_closerPeers().iter_mut() {
+                    // TODO: for now we don't parse properly the peer, so it is possible that we
+                    //       get parsing errors even though the peer is valid ; we ignore these,
+                    //       but ultimately we should just error altogether
+                    if let Ok(peer) = Peer::from_peer(peer) {
+                        closer_peers.push(peer);
+                    }
+                }
+                // TODO: restore once the above TODO is fixed
+                //debug_assert_eq!(closer_peers.len(), closer_peers.capacity());
```

Comment:
Actually when the above is fixed then you'd want to do:

```rust
message.mut_closerPeers()
    .iter_mut()
    .map(Peer::from_peer)
    .collect::<Result<Vec<_>, _>>()?
```

Also commented code